### ISSUE TITLE
Does NumReads also correct for length bias like default TPM does. 

### ISSUE THREAD

**Issue by callumparr** [COMBINE-lab/salmon#651]
https://github.com/COMBINE-lab/salmon/issues/602#issuecomment-748729130

Having read this and also wanting to use salmon to quantify transcript abundances from direct RNA data I have been using the pipeline here: https://github.com/nanoporetech/pipeline-transcriptome-de 

In this case --noErrorModel is used but not --NoLengthCorrection. The pipeline uses NumReads as the counts in downstream analysis but I was wondering if this estimation of transcript counts also tries to normalize the length bias typical of short-read RNA seq workflows? If not, I think NumReads is OK but if it does this also seems not useable like the default TPM output.

**Comment by rob-p** [comment]
Dear @callumparr,

  Thank you for bringing this up.  So you are correct that the `--noLengthCorrection` flag should be passed to salmon when quantifying data that does not have a "fragmentation effect", that is, where the number of fragments we expect to draw from a transcript is not dependent upon the length of that transcript.  In the ONT protocols, it is usually the case that we get 1 read -> 1 transcript, even if we don't read the whole thing.

  We have tested the effect of this in ONT data with spike ins, and have verified that using `--noLengthCorrection` does generally lead to improved accuracy with respect to quantification estimates.  We have informed ONT of this, and I would guess they may optimize the flags that are used soon (we have also developed an error model that works correctly for these long reads, and that should make it into the next release of salmon).

  Regarding the effect this has on the `NumReads` values reported by salmon, it's not as simple as with the `TPM` estimates.  The length affects the assigned reads through the probabilistic model on which inference is done.  With the length effect we have that P(f | t_i) ∝ P( position | f, t_i ) * P( alignment | f, t_i) --- forgetting the alignment term for the time being, we have that with length correction P( position | f, t_i ) ∝ 1 / l_i and without length correction the l_i term goes away.  In other words, the probability of allocating reads has a term that depends on the effective length when the `--noLengthCorrection` flag is not passed, but that term goes away when it is passed.  This is not quite as drastic as with TPM where the normalization includes the length directly in the normalization (note, however, that when the  `--noLengthCorrection` flag is passed, this adjusts the TPM as well).  Further, the `NumReads` is still better than TPM in this regard because it still encodes the effect size (i.e. `NumReads` will sum to the total number of aligned reads).

  Anyway TLDR: Passing the `--noLengthCorrection` flag *is* better for ONT data, though results without this flag are sub-optimal, they are not unusable.  We have let ONT know about this, and I would suspect they will address it (perhaps they'll even accept a PR?).  Finally, a long read error model has been created and will _hopefully_ make it to the next version of salmon.

Best,
Rob

**Comment by callumparr** [comment]
Thank you for your explanation. For time being I will alter my snakefile to reflect and already raised an issue with ONT
